### PR TITLE
Add dyff, a go binary.

### PR DESCRIPTION
**Pull request by asford**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by asford** [comment]
@conda-forge/help-go This is ready for review.

**Comment by asford** [comment]
@conda-forge/staged-recipes This is ready for review.

**Comment by isuruf** [comment]
See https://github.com/conda-forge/conda-forge.github.io/issues/1052

**Comment by asford** [comment]
@isuruf I've added the `go-licenses` step into the package. This includes all the required license information and/or source code required for redistribution.

I wasn't able to find any specific guidance in https://github.com/conda-forge/conda-forge.github.io/issues/1052 for how to do this, and it doesn't appear to be broadly implemented the go-related conda-forge ecosystem.

File: recipes/dyff/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - folder: {{ pkg_src }}
+    url: https://{{ goname }}/archive/v{{ version }}.tar.gz
+    sha256: 933d3545451019ccb35d799a4cf09e775b973d3b0b9ecef82c16e407ae6975c4
+
+build:
+  number: 1
```
Comment (chrisburr) [inline recipes/dyff/meta.yaml:17]: This should start from zero and only be incremented if the previous build number was uploaded to the main channel:
```suggestion
  number: 0
```

Comment (asford) [reply recipes/dyff/meta.yaml]: Roger, thanks. Ready for re-review.

**Pull request merged**